configure/CMakeLists.txt

```diff
@@ -219,6 +220,18 @@ if(NOT WIN32)
     endif()
 endif()
 
+if(TANGO_ABI_API_CHECKING)
+  if(NOT COMPILER_SUPPORTS_OG OR NOT COMPILER_SUPPORTS_G)
+    message(FATAL_ERROR "Your compiler does not support -g or -Og but this is required for ABI/API compliance checking.")
+  endif()
+
+  if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+    message(FATAL_ERROR "Requires Debug release type for ABI/API compliance checking.")
+  endif()
+
+  add_compile_options(-Og -g)
```

Comment:
How about removing TANGO_ABI_API_CHECKING options and enabling '-Og -g' for all debug builds unconditionally? Currently we enable only  '-Og' for all debug builds.

Reply:
BTW I just noticed it is compiling with '-g -Og' already:
```
[ 16%] Building CXX object cppapi/server/CMakeFiles/server_objects.dir/eventsupplier.cpp.o
cd /home/michal/Documents/cppTango/build/cppapi/server && /usr/bin/c++  -D_REENTRANT -I/home/michal/Documents/cppTango/cppapi/client -I/home/michal/Documents/cppTango/cppapi/client/helpers -
I/home/michal/Documents/cppTango/cppapi/server -I/home/michal/Documents/cppTango/cppapi/server/idl -I/home/michal/Documents/cppTango/log4tango/include -I/home/michal/Documents/cppTango/build
/cppapi/server -I/home/michal/Documents/cppTango/build/log4tango/include  -std=c++14 -g   -Og -Wall -Wextra -pedantic -fPIC -Winvalid-pch -include /home/michal/Documents/cppTango/build/cppap
i/server/tango.h -o CMakeFiles/server_objects.dir/eventsupplier.cpp.o -c /home/michal/Documents/cppTango/cppapi/server/eventsupplier.cpp
```

My configuration: ubuntu bionic (docker):
```
cmake .. -DCMAKE_INSTALL_PREFIX=/usr/local -DUSE_PCH=ON -DCMAKE_BUILD_TYPE=Debug -DIDL_BASE=$HOME/Documents/tangoid
```

Reply:
Always using `-g -Og` for debug is a good idea.